query.go

```diff
@@ -965,7 +965,7 @@ func (s *ChainService) GetBlock(blockHash chainhash.Hash,
 	inv := wire.NewInvVect(invType, &blockHash)
 
 	// If the block is already in the cache, we can return it immediately.
-	blockValue, err := s.BlockCache.Get(*inv)
+	blockValue, err := s.BlockCache.Get(blockHash)
```

Comment:
The main difference to before is that we now don't use the `invType` in the key anymore now. I'm not sure if this can have any side effects, @wpaulino? I assume not since we're only interested in witness blocks anyway. Just wanted to highlight this.

Reply:
I had suggested this change to maintain the cache keys used here and within lnd consistent. I didn't realize this would break the API however, as now the block won't include the witnesses if it was previously cached without them.

@ellemouton It seems like we'll need to introduce the inv type to lnd's cache key as well, and just always set it to `WitnessEncoding` as that's the default for `GetBlock` on other backends.

Reply:
cool, updated 👍